### PR TITLE
Enable T5 generation and beam search

### PR DESCRIPTION
**Pull request by christofid**

### CODE REVIEW

File: src/gt4sd/algorithms/generation/tests/test_hugging_face.py [hunk 1]
```diff
@@ -170,6 +179,11 @@ def test_available_versions(config_class: Type[AlgorithmConfiguration]):
             HuggingFaceGenerationAlgorithm,
             marks=pytest.mark.skipif(test_settings.gt4sd_ci, reason="high_memory"),
         ),
+        pytest.param(
+            HuggingFaceT5Generator,
+            HuggingFaceGenerationAlgorithm,
+            marks=pytest.mark.skipif(test_settings.gt4sd_ci, reason="high_memory"),
```
Comment (drugilsberg) [inline src/gt4sd/algorithms/generation/tests/test_hugging_face.py:185]: I would not skip the test

```suggestion
            marks=pytest.mark.skipif(test_settings.gt4sd_ci, reason="high_memory")
```

File: src/gt4sd/algorithms/generation/hugging_face/implementation.py [hunk 2]
```diff
@@ -137,6 +139,7 @@ def adjust_length_to_model(length: int, maximum_sequence_length: int):
     "xlnet": (XLNetLMHeadModel, XLNetTokenizer, prepare_prefix_input),
     "transfo-xl": (TransfoXLLMHeadModel, TransfoXLTokenizer, prepare_prefix_input),
     "xlm": (XLMWithLMHeadModel, XLMTokenizer, None),
+    "t5": (AutoModelForSeq2SeqLM, AutoTokenizer, None),
```
Comment (drugilsberg) [inline src/gt4sd/algorithms/generation/hugging_face/implementation.py:142]: should we consider using another key name as it seems more generic than t5, maybe auto-seq2seq-lm?

**Review by drugilsberg — CHANGES_REQUESTED** [review]
minor changes

**Comment by christofid** [comment]
Thanks @drugilsberg, I addressed these comments

**Review by drugilsberg — COMMENTED** [review]
Last small comment

File: src/gt4sd/algorithms/generation/tests/test_hugging_face.py [hunk 3]
```diff
@@ -140,36 +149,13 @@ def test_available_versions(config_class: Type[AlgorithmConfiguration]):
...
-            HuggingFaceGenerationAlgorithm,
-            marks=pytest.mark.skipif(test_settings.gt4sd_ci, reason="high_memory"),
-        ),
+        pytest.param(HuggingFaceXLMGenerator, HuggingFaceGenerationAlgorithm),
+        pytest.param(HuggingFaceCTRLGenerator, HuggingFaceGenerationAlgorithm),
+        pytest.param(HuggingFaceGPT2Generator, HuggingFaceGenerationAlgorithm),
+        pytest.param(HuggingFaceOpenAIGPTGenerator, HuggingFaceGenerationAlgorithm),
+        pytest.param(HuggingFaceXLNetGenerator, HuggingFaceGenerationAlgorithm),
+        pytest.param(HuggingFaceTransfoXLGenerator, HuggingFaceGenerationAlgorithm),
+        pytest.param(HuggingFaceSeq2SeqGenerator, HuggingFaceGenerationAlgorithm),
```
Comment (drugilsberg) [inline src/gt4sd/algorithms/generation/tests/test_hugging_face.py:158]: here I would enable only the last one (using t5-small), sorry if I confused you

File: src/gt4sd/algorithms/generation/hugging_face/implementation.py [hunk 4]
```diff
@@ -213,10 +221,14 @@ def load_model(self) -> None:
         )
         self.model = model_class.from_pretrained(model_name_or_path)
         self.model.to(self.device)
+
         # adjusting length
-        self.length = adjust_length_to_model(
-            self.length, self.model.config.max_position_embeddings
-        )
+        if self.model_type == "auto-seq2seq-lm":
+            self.length = adjust_length_to_model(self.length, 512)
```
Comment (drugilsberg) [inline src/gt4sd/algorithms/generation/hugging_face/implementation.py:227]: would it be possible to infer that 512 somehow? That hardcoding worries me.

**Review by drugilsberg — APPROVED** [review]
LGTM

**Pull request merged**